drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -370,6 +371,11 @@ static int bq274xx_gauge_init(const struct device *dev)
 	int status = 0;
 	uint16_t id;
 
+	if (!device_is_ready(config->int_gpios.port)) {
+		LOG_ERR("GPIO device pointer is not ready to be used");
+		return -EINVAL;
```

Comment:
```suggestion
		return -ENODEV;
```

Reply:
`int_gpios` is only available if PM is enabled, so it won't always compile. Is it intended to force all instances of the driver to provide this GPIO if PM is enabled?

Reply:
Yeah this GPIO should be provided only if PM is enabled, complete oversight on my part. I've wrapped it around the `#ifdef CONFIG_PM_DEVICE` check.